### PR TITLE
feat: Add pre-commit hooks and use git-crypt in the pipeline

### PR DESCRIPTION
**Pull request by kuwairo**
**Summary**

Tasks:

1. Install the tools below, learn how to use them:
- https://pre-commit.com
- https://github.com/AGWA/git-crypt
2. Add pre-commit hooks to your repo with a test task (in a separate PR), include the following rules:
- https://github.com/adrienverge/yamllint.git
- https://github.com/lyz-code/yamlfix
- https://github.com/shellcheck-py/shellcheck-py
- https://github.com/petalmd/git-crypt-pre-commit (ensure git-crypt is installed)
- https://github.com/pre-commit/pre-commit-hooks (check-added-large-files, check-merge-conflict, forbid-new-submodules, no-commit-to-branch, check-symlinks, mixed-line-ending, end-of-file-fixer, trailing-whitespace)
3. Refactor your solution to take secret parameters during CI from a file encrypted by git-crypt.
4. Open a new PR to your repository with your test task and invite Kseniya when the task is complete.

### CODE REVIEW

File: .git-crypt/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Do not edit this file.  To specify the files to encrypt, create your own
```
Comment (kseniyashaydurova) [inline .git-crypt/.gitattributes]: Let's add whole content of `.git-crypt` folder to `.gitignore`, as well as below ones:
- .idea
- .vscode
- git-crypt-key

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        args: [--config-file, .yamlfix.toml]
+  - repo: https://github.com/shellcheck-py/shellcheck-py
+    rev: v0.10.0.1
+    hooks:
+      - id: shellcheck
+  - repo: https://github.com/petalmd/git-crypt-pre-commit
+    rev: v0.1
+    hooks:
+      - id: git-crypt
+  - repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (kseniyashaydurova) [inline .pre-commit-config.yaml:18]: Let's order all our hooks and repos in ASC order

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+---
+repos:
+  - repo: https://github.com/adrienverge/yamllint
```
Comment (kseniyashaydurova) [inline .pre-commit-config.yaml:2]: Let's separate repos items for better readability?

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    rev: v1.35.1
+    hooks:
+      - id: yamllint
```
Comment (kseniyashaydurova) [inline .pre-commit-config.yaml:5]: Let's feed config to our yamllint with below content:

```sh
extends: default

yaml-files:
  - '*.yaml'
  - '*.yml'
  - '.yamllint'
  -
# see rules
# https://yamllint.readthedocs.io/en/stable/rules.html
rules:
  line-length: disable
  document-start: disable
  comments: disable
```

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  - repo: https://github.com/lyz-code/yamlfix
+    rev: 1.16.1
+    hooks:
+      - id: yamlfix
+        args: [--config-file, .yamlfix.toml]
+  - repo: https://github.com/shellcheck-py/shellcheck-py
+    rev: v0.10.0.1
+    hooks:
+      - id: shellcheck
```
Comment (kseniyashaydurova) [inline .pre-commit-config.yaml]: Let's find a param that we can pass to shellcheck to check also attached `sources` to bash files - https://github.com/koalaman/shellcheck

File: docker-compose.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
+---
 services:
   app:
     image: ghcr.io/kuwairo/angular-react-starter:latest
     restart: on-failure
-    ports:
-    - 127.0.0.1:8087:8080
+    ports: [127.0.0.1:8087:8080]
```
Comment (kseniyashaydurova) [inline docker-compose.yaml]: Why it fixed by yamlfix in such a way? Can we change it to:

```
ports:
  - "8087:8080"
```

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+---
+repos:
+  - repo: https://github.com/adrienverge/yamllint
+    rev: v1.35.1
+    hooks:
+      - id: yamllint
+  - repo: https://github.com/lyz-code/yamlfix
+    rev: 1.16.1
```
Comment (kseniyashaydurova) [inline .pre-commit-config.yaml]: Let's use latest versions everywhere?

File: .git-crypt/.gitattributes (hunk 1 above)
Comment (kuwairo) [reply .git-crypt/.gitattributes]: Done

File: .pre-commit-config.yaml (hunk 2 above)
Comment (kuwairo) [reply .pre-commit-config.yaml:18]: Done

File: .pre-commit-config.yaml (hunk 3 above)
Comment (kuwairo) [reply .pre-commit-config.yaml:2]: Done

File: .pre-commit-config.yaml (hunk 4 above)
Comment (kuwairo) [reply .pre-commit-config.yaml:5]: Done

File: .pre-commit-config.yaml (hunk 5 above)
Comment (kuwairo) [reply .pre-commit-config.yaml]: Done

File: docker-compose.yaml (hunk 6 above)
Comment (kuwairo) [reply docker-compose.yaml]: Done

File: .pre-commit-config.yaml (hunk 7 above)
Comment (kuwairo) [reply .pre-commit-config.yaml]: Done

**Review by kseniyashaydurova — APPROVED** [review]

**Pull request merged**